src/clj_sqs_extended/core.clj

```diff
@@ -72,43 +73,58 @@
   (.deleteQueue sqs-client url))
 
 (defn send-message
-  [sqs-client url message]
-  (.sendMessage sqs-client url message))
+  ([sqs-client url body]
+   (send-message sqs-client url body {}))
+  ([sqs-client url body
+    {:keys [format]
+     :or   {format :transit}
+     :as   opts}]
+   (let [payload (serdes/serialize body format)]
+     (.sendMessage sqs-client url payload))))
 
 (defn send-fifo-message
-  [sqs-client url message group-id]
-  (let [request (SendMessageRequest. url message)]
-    ; WATCHOUT: The group ID is mandatory when sending fifo messages.
-    (doto request (.setMessageGroupId group-id))
-    (.sendMessage sqs-client request)))
+  ([sqs-client url body group-id]
+   (send-fifo-message sqs-client url body group-id {}))
+  ([sqs-client url body group-id
+    {:keys [format]
+     :or   {format :transit}
+     :as   opts}]
+   (let [payload (serdes/serialize body format)
+         request (SendMessageRequest. url payload)]
+     ; WATCHOUT: The group ID is mandatory when sending fifo messages.
+     (doto request (.setMessageGroupId group-id))
+     (.sendMessage sqs-client request))))
 
 (defn delete-message
-  [sqs-client url message]
-  (let [request (DeleteMessageRequest. url (:receiptHandle message))]
+  [sqs-client url handle]
```

Comment:
why is it called `handle`? it's not clear to the user what is a handle. I can see that you're extracting the `handle` from a mesage. But what's get passed is a message. And calling it handle is leaking implementation to the user.

lastly, there's an obvious pattern to the API signature. `send-mesage(client url message)`, so we should stick with `delete-message(client url message)` for consistency of the API terminology for user's sake.

Reply:
Stupid miss.